% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{article}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage[margin=1in]{geometry}
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{-\maxdimen} % remove section numbering
\ifluatex
  \usepackage{selnolig}  % disable illegal ligatures
\fi

\author{}
\date{\vspace{-2.5em}}

\begin{document}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}

title: ``R\_Assignment'' author: ``Kavi R. Acharya'' date: ``March 19,
2021'' output: word\_document: default pdf\_document: default
html\_document: default

\hypertarget{part-i-unix-assignment-in-r}{%
\subsection{Part I: Unix assignment in
R}\label{part-i-unix-assignment-in-r}}

Loading the tidyverse package:

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{require}\NormalTok{(}\StringTok{"tidyverse"}\NormalTok{)) }\FunctionTok{install.packages}\NormalTok{(}\StringTok{"tidyverse"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Loading required package: tidyverse
\end{verbatim}

\begin{verbatim}
## Warning: package 'tidyverse' was built under R version 3.6.3
\end{verbatim}

\begin{verbatim}
## -- Attaching packages --------------------------------------- tidyverse 1.3.0 --
\end{verbatim}

\begin{verbatim}
## v ggplot2 3.3.3     v purrr   0.3.4
## v tibble  3.1.0     v dplyr   1.0.4
## v tidyr   1.1.2     v stringr 1.4.0
## v readr   1.4.0     v forcats 0.5.1
\end{verbatim}

\begin{verbatim}
## Warning: package 'ggplot2' was built under R version 3.6.3
\end{verbatim}

\begin{verbatim}
## Warning: package 'tibble' was built under R version 3.6.3
\end{verbatim}

\begin{verbatim}
## Warning: package 'tidyr' was built under R version 3.6.3
\end{verbatim}

\begin{verbatim}
## Warning: package 'readr' was built under R version 3.6.3
\end{verbatim}

\begin{verbatim}
## Warning: package 'purrr' was built under R version 3.6.3
\end{verbatim}

\begin{verbatim}
## Warning: package 'dplyr' was built under R version 3.6.3
\end{verbatim}

\begin{verbatim}
## Warning: package 'stringr' was built under R version 3.6.3
\end{verbatim}

\begin{verbatim}
## Warning: package 'forcats' was built under R version 3.6.3
\end{verbatim}

\begin{verbatim}
## -- Conflicts ------------------------------------------ tidyverse_conflicts() --
## x dplyr::filter() masks stats::filter()
## x dplyr::lag()    masks stats::lag()
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(tidyverse)}
\end{Highlighting}
\end{Shaded}

\hypertarget{downloading-the-fang-et-al-and-snp-files}{%
\subsection{Downloading the Fang et al and snp
files}\label{downloading-the-fang-et-al-and-snp-files}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Downloading files directly from the github repository }
\NormalTok{snp}\OtherTok{\textless{}{-}}\FunctionTok{read\_tsv}\NormalTok{(}\StringTok{"https://raw.githubusercontent.com/EEOB{-}BioData/BCB546{-}Spring2021/main/assignments/UNIX\_Assignment/snp\_position.txt"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## -- Column specification --------------------------------------------------------
## cols(
##   SNP_ID = col_character(),
##   cdv_marker_id = col_double(),
##   Chromosome = col_character(),
##   Position = col_character(),
##   alt_pos = col_character(),
##   mult_positions = col_character(),
##   amplicon = col_character(),
##   cdv_map_feature.name = col_character(),
##   gene = col_character(),
##   `candidate/random` = col_character(),
##   Genaissance_daa_id = col_double(),
##   Sequenom_daa_id = col_double(),
##   count_amplicons = col_double(),
##   count_cmf = col_double(),
##   count_gene = col_double()
## )
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Fang}\OtherTok{\textless{}{-}}\FunctionTok{read\_tsv}\NormalTok{(}\StringTok{"https://raw.githubusercontent.com/EEOB{-}BioData/BCB546{-}Spring2021/main/assignments/UNIX\_Assignment/fang\_et\_al\_genotypes.txt"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## 
## -- Column specification --------------------------------------------------------
## cols(
##   .default = col_character()
## )
## i Use `spec()` for the full column specifications.
\end{verbatim}

\hypertarget{data-inspection}{%
\subsection{Data Inspection}\label{data-inspection}}

we can inspect the data fang et al as well as snp datafile using
following

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Load data:}
\FunctionTok{str}\NormalTok{(snp)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## spec_tbl_df [983 x 15] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
##  $ SNP_ID              : chr [1:983] "abph1.20" "abph1.22" "ae1.3" "ae1.4" ...
##  $ cdv_marker_id       : num [1:983] 5976 5978 6605 6606 6607 ...
##  $ Chromosome          : chr [1:983] "2" "2" "5" "5" ...
##  $ Position            : chr [1:983] "27403404" "27403892" "167889790" "167889682" ...
##  $ alt_pos             : chr [1:983] NA NA NA NA ...
##  $ mult_positions      : chr [1:983] NA NA NA NA ...
##  $ amplicon            : chr [1:983] "abph1" "abph1" "ae1" "ae1" ...
##  $ cdv_map_feature.name: chr [1:983] "AB042260" "AB042260" "ae1" "ae1" ...
##  $ gene                : chr [1:983] "abph1" "abph1" "ae1" "ae1" ...
##  $ candidate/random    : chr [1:983] "candidate" "candidate" "candidate" "candidate" ...
##  $ Genaissance_daa_id  : num [1:983] 8393 8394 8395 8396 8397 ...
##  $ Sequenom_daa_id     : num [1:983] 10474 10475 10477 10478 10479 ...
##  $ count_amplicons     : num [1:983] 1 0 1 0 0 1 1 0 1 0 ...
##  $ count_cmf           : num [1:983] 1 0 1 0 0 1 0 0 1 0 ...
##  $ count_gene          : num [1:983] 1 0 1 0 0 1 1 0 1 0 ...
##  - attr(*, "spec")=
##   .. cols(
##   ..   SNP_ID = col_character(),
##   ..   cdv_marker_id = col_double(),
##   ..   Chromosome = col_character(),
##   ..   Position = col_character(),
##   ..   alt_pos = col_character(),
##   ..   mult_positions = col_character(),
##   ..   amplicon = col_character(),
##   ..   cdv_map_feature.name = col_character(),
##   ..   gene = col_character(),
##   ..   `candidate/random` = col_character(),
##   ..   Genaissance_daa_id = col_double(),
##   ..   Sequenom_daa_id = col_double(),
##   ..   count_amplicons = col_double(),
##   ..   count_cmf = col_double(),
##   ..   count_gene = col_double()
##   .. )
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{unique}\NormalTok{(Fang}\SpecialCharTok{$}\NormalTok{Group)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "TRIPS" "ZDIPL" "ZPERR" "ZLUXR" "ZMHUE" "ZMPBA" "ZMPJA" "ZMXCH" "ZMXCP"
## [10] "ZMXNO" "ZMXNT" "ZMPIL" "ZMXIL" "ZMMLR" "ZMMMR" "ZMMIL"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{unique}\NormalTok{(snp}\SpecialCharTok{$}\NormalTok{Chromosome)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
##  [1] "2"        "5"        "1"        "3"        "4"        "6"       
##  [7] "9"        "8"        "multiple" "7"        "unknown"  "10"
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{nrow}\NormalTok{(snp)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 983
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{ncol}\NormalTok{(snp)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## [1] 15
\end{verbatim}

\hypertarget{data-processing}{%
\subsubsection{Data Processing}\label{data-processing}}

At first we need to rearrange the snp file. SNP file is rearragned so
that Chromosome is kept at second column followed by the position. Then
the file is transposed.

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Rearranging the snp file}
\NormalTok{snp}\OtherTok{\textless{}{-}}\NormalTok{snp[}\FunctionTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{,}\DecValTok{3}\NormalTok{,}\DecValTok{4}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{5}\SpecialCharTok{:}\DecValTok{15}\NormalTok{)]}

\CommentTok{\#Creating subsets of Maize and Teosinte genotype files}

\NormalTok{Maize.genotype}\OtherTok{\textless{}{-}}\NormalTok{Fang }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(Group}\SpecialCharTok{==}\StringTok{"ZMMIL"}\SpecialCharTok{|}\NormalTok{Group}\SpecialCharTok{==}\StringTok{"ZMMLR"}\SpecialCharTok{|}\NormalTok{Group}\SpecialCharTok{==}\StringTok{"ZMMMR"}\NormalTok{) }

\NormalTok{Teosinte.genotype}\OtherTok{\textless{}{-}}\NormalTok{Fang }\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{filter}\NormalTok{(Group}\SpecialCharTok{==}\StringTok{"ZMPJA"}\SpecialCharTok{|}\NormalTok{Group}\SpecialCharTok{==}\StringTok{"ZMPIL"}\SpecialCharTok{|}\NormalTok{Group}\SpecialCharTok{==}\StringTok{"ZMPBA"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{now-we-transpose-the-subsets-and-merge.}{%
\subsubsection{Now, we transpose the subsets and
merge.}\label{now-we-transpose-the-subsets-and-merge.}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Maize.genotype}\OtherTok{\textless{}{-}}\FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{t}\NormalTok{(Maize.genotype), }\AttributeTok{stringsAsFactors =}\NormalTok{ F)}
\NormalTok{Teosinte.genotype}\OtherTok{\textless{}{-}}\FunctionTok{as.data.frame}\NormalTok{(}\FunctionTok{t}\NormalTok{(Teosinte.genotype), }\AttributeTok{stringsAsFactors =}\NormalTok{ F)}

\CommentTok{\#Here, we are transposing column name intp row name. Then, a new row name is created under SNP\_ID. The first row is converted into row name.At last, first three columns are removed.}
\NormalTok{SNP\_ID }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(Maize.genotype)}
\FunctionTok{rownames}\NormalTok{(Maize.genotype) }\OtherTok{\textless{}{-}} \ConstantTok{NULL}
\NormalTok{Maize.genotype }\OtherTok{\textless{}{-}} \FunctionTok{cbind}\NormalTok{(SNP\_ID,Maize.genotype,}\AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}\NormalTok{)}
\FunctionTok{names}\NormalTok{(Maize.genotype)}\OtherTok{\textless{}{-}}  \FunctionTok{c}\NormalTok{(}\StringTok{"SNP\_ID"}\NormalTok{,Maize.genotype[}\DecValTok{1}\NormalTok{,}\SpecialCharTok{{-}}\DecValTok{1}\NormalTok{])}
\NormalTok{Maize.genotype }\OtherTok{\textless{}{-}}\NormalTok{ Maize.genotype[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{3}\NormalTok{), ]}


\NormalTok{SNP\_ID }\OtherTok{\textless{}{-}} \FunctionTok{rownames}\NormalTok{(Teosinte.genotype)}
\FunctionTok{rownames}\NormalTok{(Teosinte.genotype) }\OtherTok{\textless{}{-}} \ConstantTok{NULL}
\NormalTok{Teosinte.genotype }\OtherTok{\textless{}{-}} \FunctionTok{cbind}\NormalTok{(SNP\_ID,Teosinte.genotype,}\AttributeTok{stringsAsFactors =} \ConstantTok{FALSE}\NormalTok{)}
\FunctionTok{names}\NormalTok{(Teosinte.genotype)}\OtherTok{\textless{}{-}}  \FunctionTok{c}\NormalTok{(}\StringTok{"SNP\_ID"}\NormalTok{,Teosinte.genotype[}\DecValTok{1}\NormalTok{,}\SpecialCharTok{{-}}\DecValTok{1}\NormalTok{])}
\NormalTok{Teosinte.genotype }\OtherTok{\textless{}{-}}\NormalTok{ Teosinte.genotype[}\SpecialCharTok{{-}}\FunctionTok{c}\NormalTok{(}\DecValTok{1}\NormalTok{,}\DecValTok{2}\NormalTok{,}\DecValTok{3}\NormalTok{), ]}
\end{Highlighting}
\end{Shaded}

Now, we are going to merge the transposed genotype file with snp files

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Merged.maize}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(snp, Maize.genotype, }\AttributeTok{by=}\StringTok{"SNP\_ID"}\NormalTok{)}
\NormalTok{Merged.teosinte}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(snp, Teosinte.genotype, }\AttributeTok{by=}\StringTok{"SNP\_ID"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

We now subset the merged dataset based on Chromosome number, sort it
based on position and export it in the form of csv files

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#Here, the as.numeric function will give us an error message when there is "unknow/multiple" value. We will replace such value by NA. New column Pos is created and the data is sorted numerically. After sorting, Pos is removed. }

\ControlFlowTok{for}\NormalTok{(i }\ControlFlowTok{in} \FunctionTok{c}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\DecValTok{10}\NormalTok{))\{}
\NormalTok{  data}\OtherTok{\textless{}{-}}\NormalTok{Merged.maize }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(Chromosome}\SpecialCharTok{==}\NormalTok{i)}\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{mutate}\NormalTok{(}\AttributeTok{Pos=}\FunctionTok{as.numeric}\NormalTok{(Position))}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{arrange}\NormalTok{(Pos)}
\NormalTok{  data}\SpecialCharTok{$}\NormalTok{Pos}\OtherTok{\textless{}{-}}\ConstantTok{NULL}
  \FunctionTok{write.csv}\NormalTok{( data,}\FunctionTok{paste0}\NormalTok{(}\StringTok{"Maize\_Chromo\_"}\NormalTok{,i,}\StringTok{"\_ascending.csv"}\NormalTok{), }\AttributeTok{row.names =}\NormalTok{ F)}
\NormalTok{  data}\OtherTok{\textless{}{-}}\NormalTok{Merged.maize}\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(Chromosome}\SpecialCharTok{==}\NormalTok{i)}\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{mutate}\NormalTok{(}\AttributeTok{Pos=}\FunctionTok{as.numeric}\NormalTok{(Position))}\SpecialCharTok{\%\textgreater{}\%}
    \FunctionTok{arrange}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{Pos)}
\NormalTok{  data}\SpecialCharTok{$}\NormalTok{Pos}\OtherTok{\textless{}{-}}\ConstantTok{NULL}
\NormalTok{  data[data}\SpecialCharTok{==}\StringTok{"?/?"}\NormalTok{]}\OtherTok{\textless{}{-}}\StringTok{"{-}/{-}"}
  \FunctionTok{write.csv}\NormalTok{( data,}\FunctionTok{paste0}\NormalTok{(}\StringTok{"Maize\_Chromo\_"}\NormalTok{,i,}\StringTok{"\_descending.csv"}\NormalTok{), }\AttributeTok{row.names =}\NormalTok{ F)}
  
\NormalTok{  data}\OtherTok{\textless{}{-}}\NormalTok{Merged.teosinte }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(Chromosome}\SpecialCharTok{==}\NormalTok{i)}\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{mutate}\NormalTok{(}\AttributeTok{Pos=}\FunctionTok{as.numeric}\NormalTok{(Position))}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{arrange}\NormalTok{(Pos)}
\NormalTok{  data}\SpecialCharTok{$}\NormalTok{Pos}\OtherTok{\textless{}{-}}\ConstantTok{NULL}
  \FunctionTok{write.csv}\NormalTok{( data,}\FunctionTok{paste0}\NormalTok{(}\StringTok{"Teosinte\_Chromo\_"}\NormalTok{,i,}\StringTok{"\_ascending.csv"}\NormalTok{), }\AttributeTok{row.names =}\NormalTok{ F)}
\NormalTok{  data}\OtherTok{\textless{}{-}}\NormalTok{Merged.teosinte}\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(Chromosome}\SpecialCharTok{==}\NormalTok{i)}\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{mutate}\NormalTok{(}\AttributeTok{Pos=}\FunctionTok{as.numeric}\NormalTok{(Position))}\SpecialCharTok{\%\textgreater{}\%}
    \FunctionTok{arrange}\NormalTok{(}\SpecialCharTok{{-}}\NormalTok{Pos)}
\NormalTok{  data}\SpecialCharTok{$}\NormalTok{Pos}\OtherTok{\textless{}{-}}\ConstantTok{NULL}
\NormalTok{  data[data}\SpecialCharTok{==}\StringTok{"?/?"}\NormalTok{]}\OtherTok{\textless{}{-}}\StringTok{"{-}/{-}"}
  \FunctionTok{write.csv}\NormalTok{( data,}\FunctionTok{paste0}\NormalTok{(}\StringTok{"Teosinte\_Chromo\_"}\NormalTok{,i,}\StringTok{"\_descending.csv"}\NormalTok{), }\AttributeTok{row.names =}\NormalTok{ F)}
  
\NormalTok{\}}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion

## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion
\end{verbatim}

\hypertarget{part-ii}{%
\section{Part II}\label{part-ii}}

Data Visualization.

\hypertarget{snps-per-chromosome}{%
\subsubsection{SNPs per chromosome}\label{snps-per-chromosome}}

We are going to visualize the number of possible polymorphism in each
chromosome.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(ggplot2)}

\FunctionTok{ggplot}\NormalTok{(}\AttributeTok{data =}\NormalTok{ snp[}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(snp}\SpecialCharTok{$}\NormalTok{Chromosome)),]) }\SpecialCharTok{+}   \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{mapping =} \FunctionTok{aes}\NormalTok{(}\AttributeTok{x =} \FunctionTok{as.numeric}\NormalTok{(Chromosome), }\AttributeTok{fill=}\NormalTok{Chromosome)) }\SpecialCharTok{+} \FunctionTok{scale\_x\_discrete}\NormalTok{(}\AttributeTok{limit=}\FunctionTok{c}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\DecValTok{10}\NormalTok{))}\SpecialCharTok{+} \FunctionTok{labs}\NormalTok{(}\AttributeTok{x =} \StringTok{"Chromosome number"}\NormalTok{, }\AttributeTok{y=}\StringTok{"No. of polymorphism position"}\NormalTok{) }
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in `[.tbl_df`(snp, !is.na(as.numeric(snp$Chromosome)), ): NAs introduced
## by coercion
\end{verbatim}

\begin{verbatim}
## Warning: Continuous limits supplied to discrete scale.
## Did you mean `limits = factor(...)` or `scale_*_continuous()`?
\end{verbatim}

\includegraphics{Assignment-Workflow_files/figure-latex/unnamed-chunk-8-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#Traforming the data using pivot\_longer. }

\NormalTok{Pivot}\OtherTok{\textless{}{-}}\NormalTok{Fang }\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{pivot\_longer}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{c}\NormalTok{(Sample\_ID, JG\_OTU, Group),}\AttributeTok{names\_to=}\StringTok{"SNP\_ID"}\NormalTok{,}\AttributeTok{values\_to=} \StringTok{"Base"}\NormalTok{)}
\NormalTok{Pivot}\OtherTok{\textless{}{-}}\FunctionTok{merge}\NormalTok{(Pivot, snp, }\AttributeTok{by=}\StringTok{"SNP\_ID"}\NormalTok{)}

\FunctionTok{ggplot}\NormalTok{(}\AttributeTok{data =}\NormalTok{ Pivot[}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(Pivot}\SpecialCharTok{$}\NormalTok{Chromosome)),]) }\SpecialCharTok{+}   \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{mapping =} \FunctionTok{aes}\NormalTok{(  }\FunctionTok{as.numeric}\NormalTok{(Chromosome), }\AttributeTok{fill=}\NormalTok{Group)) }\SpecialCharTok{+} \FunctionTok{scale\_x\_discrete}\NormalTok{(}\AttributeTok{limit=}\FunctionTok{c}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\DecValTok{10}\NormalTok{))}\SpecialCharTok{+} \FunctionTok{labs}\NormalTok{(}\AttributeTok{x =} \StringTok{"Chromosome number"}\NormalTok{, }\AttributeTok{y=}\StringTok{"No. of SNPS"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in `[.data.frame`(Pivot, !is.na(as.numeric(Pivot$Chromosome)), ): NAs
## introduced by coercion
\end{verbatim}

\begin{verbatim}
## Warning: Continuous limits supplied to discrete scale.
## Did you mean `limits = factor(...)` or `scale_*_continuous()`?
\end{verbatim}

\includegraphics{Assignment-Workflow_files/figure-latex/unnamed-chunk-9-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#To see polymorphism in each group}

\NormalTok{Group\_graph }\OtherTok{\textless{}{-}}\NormalTok{ Pivot }\SpecialCharTok{\%\textgreater{}\%} 
  \FunctionTok{mutate}\NormalTok{(}\AttributeTok{Chromosome=}\FunctionTok{as.numeric}\NormalTok{(Chromosome)) }\SpecialCharTok{\%\textgreater{}\%} \CommentTok{\#This will mutate Chromosone number into numeric value}
  \FunctionTok{select}\NormalTok{(Group, SNP\_ID, Chromosome, Base) }\SpecialCharTok{\%\textgreater{}\%}  \CommentTok{\#this function will Select required fields}
  \FunctionTok{filter}\NormalTok{(Chromosome }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\DecValTok{10}\NormalTok{)) }\SpecialCharTok{\%\textgreater{}\%}  \CommentTok{\#Filters chromosome 1 to 10}
  \FunctionTok{filter}\NormalTok{(Base}\SpecialCharTok{!=}\StringTok{"?/?"}\NormalTok{)}\SpecialCharTok{\%\textgreater{}\%}   \CommentTok{\#Removes all unknown bases from the file}
  \FunctionTok{group\_by}\NormalTok{(Group,SNP\_ID) }\SpecialCharTok{\%\textgreater{}\%}  \CommentTok{\#Grouping by Group and SNP\_ID}
  \FunctionTok{filter}\NormalTok{(}\FunctionTok{length}\NormalTok{(}\FunctionTok{unique}\NormalTok{(Base))}\SpecialCharTok{\textgreater{}}\DecValTok{1}\NormalTok{) }\SpecialCharTok{\%\textgreater{}\%}   \CommentTok{\#This removes all SNPs with one base}
  \FunctionTok{select}\NormalTok{(Group, SNP\_ID, Chromosome) }\CommentTok{\#Selecting three fields}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in mask$eval_all_mutate(quo): NAs introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Group\_graph}\OtherTok{\textless{}{-}}\NormalTok{Group\_graph[}\SpecialCharTok{!}\FunctionTok{duplicated}\NormalTok{(Group\_graph),] }\CommentTok{\#Getting rid of duplication}

\FunctionTok{ggplot}\NormalTok{(}\AttributeTok{data=}\NormalTok{Group\_graph) }\SpecialCharTok{+}
  \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{mapping=}\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{Chromosome, }\AttributeTok{fill=}\NormalTok{Group)) }\SpecialCharTok{+} 
  \FunctionTok{scale\_x\_discrete}\NormalTok{(}\AttributeTok{limit=}\FunctionTok{c}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\DecValTok{10}\NormalTok{), }\AttributeTok{label=}\FunctionTok{c}\NormalTok{(}\DecValTok{1}\SpecialCharTok{:}\DecValTok{10}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning: Continuous limits supplied to discrete scale.
## Did you mean `limits = factor(...)` or `scale_*_continuous()`?
\end{verbatim}

\includegraphics{Assignment-Workflow_files/figure-latex/unnamed-chunk-10-1.pdf}
Converting the data into long format

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\#New column is created on the basis of homozygosity.}

\NormalTok{Pivot}\SpecialCharTok{$}\NormalTok{homozygous}\OtherTok{\textless{}{-}}\StringTok{"Heterozygous"}
\NormalTok{Pivot}\SpecialCharTok{$}\NormalTok{homozygous[Pivot}\SpecialCharTok{$}\NormalTok{Base}\SpecialCharTok{==}\StringTok{"?/?"}\NormalTok{]}\OtherTok{\textless{}{-}}\StringTok{"Missing"}
\NormalTok{Pivot}\SpecialCharTok{$}\NormalTok{homozygous[Pivot}\SpecialCharTok{$}\NormalTok{Base }\SpecialCharTok{\%in\%} \FunctionTok{c}\NormalTok{(}\StringTok{"A/A"}\NormalTok{,}\StringTok{"C/C"}\NormalTok{, }\StringTok{"G/G"}\NormalTok{, }\StringTok{"T/T"}\NormalTok{)]}\OtherTok{\textless{}{-}}\StringTok{"Homozygous"}

\CommentTok{\# SNPs are grpahed by Sample\_ID, filled by Homozygosity:}
\FunctionTok{ggplot}\NormalTok{(}\AttributeTok{data=}\NormalTok{Pivot) }\SpecialCharTok{+}
  \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{mapping=}\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{Sample\_ID, }\AttributeTok{fill=}\NormalTok{homozygous), }\AttributeTok{position=}\StringTok{"fill"}\NormalTok{) }
\end{Highlighting}
\end{Shaded}

\includegraphics{Assignment-Workflow_files/figure-latex/unnamed-chunk-11-1.pdf}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Graph the SNPs by Group, filling by Homozygosity:}
\FunctionTok{ggplot}\NormalTok{(}\AttributeTok{data=}\NormalTok{Pivot) }\SpecialCharTok{+}
  \FunctionTok{geom\_bar}\NormalTok{(}\AttributeTok{mapping=}\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{Group, }\AttributeTok{fill=}\NormalTok{homozygous), }\AttributeTok{position=}\StringTok{"fill"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Assignment-Workflow_files/figure-latex/unnamed-chunk-11-2.pdf}

To see polymorphism in each group in each chromosome.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Poly}\OtherTok{\textless{}{-}}\NormalTok{Pivot[}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(Pivot}\SpecialCharTok{$}\NormalTok{Chromosome)),]}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in `[.data.frame`(Pivot, !is.na(as.numeric(Pivot$Chromosome)), ): NAs
## introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Poly}\SpecialCharTok{$}\NormalTok{Chromosome}\OtherTok{\textless{}{-}}\FunctionTok{as.numeric}\NormalTok{(Poly}\SpecialCharTok{$}\NormalTok{Chromosome)}
\NormalTok{Poly}\OtherTok{\textless{}{-}}\NormalTok{Poly}\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{select}\NormalTok{(SNP\_ID, Group, Base, Position, Chromosome, homozygous)}\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{filter}\NormalTok{(Base}\SpecialCharTok{!=}\StringTok{"?/?"}\NormalTok{)}\SpecialCharTok{\%\textgreater{}\%} \FunctionTok{unique}\NormalTok{()}\SpecialCharTok{\%\textgreater{}\%}\FunctionTok{filter}\NormalTok{(}\SpecialCharTok{!}\FunctionTok{is.na}\NormalTok{(}\FunctionTok{as.numeric}\NormalTok{(Position)))}
\end{Highlighting}
\end{Shaded}

\begin{verbatim}
## Warning in mask$eval_all_filter(dots, env_filter): NAs introduced by coercion
\end{verbatim}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{Graph}\OtherTok{\textless{}{-}}\FunctionTok{ggplot}\NormalTok{(}\AttributeTok{data =}\NormalTok{ Poly) }\SpecialCharTok{+} \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{mapping=}\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\FunctionTok{as.numeric}\NormalTok{(Position), }\AttributeTok{y=}\NormalTok{Group, }\AttributeTok{color=}\NormalTok{homozygous)) }\SpecialCharTok{+}\FunctionTok{labs}\NormalTok{(}\AttributeTok{y =} \StringTok{"Groups"}\NormalTok{ , }\AttributeTok{x=}\StringTok{"Chromosome position"}\NormalTok{)}

\NormalTok{Graph }\SpecialCharTok{+} \FunctionTok{facet\_wrap}\NormalTok{(}\SpecialCharTok{\textasciitilde{}}\NormalTok{ Chromosome,}\AttributeTok{ncol=}\DecValTok{2}\NormalTok{,}\AttributeTok{scales =} \StringTok{"free"}\NormalTok{) }\SpecialCharTok{+} \FunctionTok{labs}\NormalTok{(}\AttributeTok{color=}\StringTok{"Genotype"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\includegraphics{Assignment-Workflow_files/figure-latex/unnamed-chunk-12-1.pdf}

\end{document}
